mlflow/pyfunc/__init__.py

```diff
@@ -1648,12 +1662,158 @@ def _convert_struct_values(
+    env_root_dir = os.path.join(_PREBUILD_ENV_ROOT_LOCATION, archive_name)
+    archive_path = os.path.join(save_path, archive_name + ".tar.gz")
+    if os.path.exists(env_root_dir):
+        shutil.rmtree(env_root_dir)
+    if os.path.exists(archive_path):
+        os.remove(archive_path)
+
     try:
-        from pyspark.sql.utils import is_remote
-    except ImportError:
-        return False
-    return is_remote()
+        pyfunc_backend = get_flavor_backend(
+            local_model_path,
+            env_manager="virtualenv",
+            install_mlflow=False,
+            create_env_root_dir=False,
+            env_root_dir=env_root_dir,
+        )
+
+        pyfunc_backend.prepare_env(model_uri=local_model_path, capture_output=False)
+        # exclude pip cache from the archive file.
+        shutil.rmtree(os.path.join(env_root_dir, "pip_cache_pkgs"))
+
+        return archive_directory(env_root_dir, archive_path)
+    finally:
+        shutil.rmtree(env_root_dir, ignore_errors=True)
+
+
+def prebuild_model_env(model_uri, save_path):
+    """
+    Prebuild model python environment and generate an archive file saved to provided
+    `save_path`.
+
+    Note:
+    You can only call `prebuild_model_env` in Databricks runtime,
+    and you can use the generated prebuilt env file in `mlflow.pyfunc.spark_udf` in
+    Databricks runtime or Databricks Connect client.
+    The generated model env file is named by:
+    'mlflow-{sha of python env config and dependencies}-{runtime version}-{platform machine}.tar.gz'
+    the file name shouldn't be modified,
+    and the prebuilt env can't be used across different Databricks runtime version or
+    different platform machine.
+    When you use the prebuilt env in other execution environment, MLflow will verify whether the
+    spark UDF sandbox environment matches the prebuilt env requirements.
+    """
+    from mlflow.utils._spark_utils import _get_active_spark_session
+
+    if not is_in_databricks_runtime():
+        raise RuntimeError("'prebuild_model_env' only support running in Databricks runtime.")
+
+    if not os.path.exists(save_path):
+        raise RuntimeError(f"The saving path '{save_path}' does not exist.")
+    if not os.path.isdir(save_path):
+        raise RuntimeError(f"The saving path '{save_path}' must be a directory.")
+
+    local_model_path = _download_artifact_from_uri(
+        artifact_uri=model_uri, output_path=_create_model_downloading_tmp_dir(should_use_nfs=False)
+    )
+    archive_name = _gen_prebuilt_env_archive_name(_get_active_spark_session(), local_model_path)
+    dest_path = os.path.join(save_path, archive_name + ".tar.gz")
+    if os.path.exists(dest_path):
+        raise RuntimeError(
+            "You have pre-built the model python environment and save "
+            f"it in the '{save_path}' directory as the archive file "
+            f"{archive_name}.tar.gz, if you want to rebuild it, please remove "
```

Comment:
```suggestion
            f"{archive_name}.tar.gz, To rebuild it, please remove "
```

"If you want" or `If you would like to` can be removed in most cases.